enhancements/installer/compact-plus-worker.md

```diff
@@ -0,0 +1,194 @@
+---
+title: Compact + Worker deployment as day0 operation
+authors:
+  - "@flaper87"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+api-approvers: # in case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers)
+  - TBD
+creation-date: 2022-03-08
+last-updated: 2022-03-08
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - TBD
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Compact + Worker deployment as day0 operation
+
+## Summary
+
+This enhancement proposes to allow for deploying a 4-nodes cluster as a day 0
+operation, where the infrastructure workload can be deployed with HA enabled.
+The 4-nodes cluster would be composed by 3 schedulable master nodes and a
+single worker, which provide users with a highly available cluster.
+
+## Motivation
+
+There are environments where only 4 nodes are available where 4-nodes clusters
+are being deployed. We could serve OpenShift cluster administrators of these
+environments better by allowing them to deploy a 4-nodes cluster as a day0
+operation and make sure their infrastructure workload is highly available.
+
+Today, it is possible to deploy a 4-nodes cluster. The resulting cluster will
+be functional but the infrastructure workload will be run as a SingleReplica
+topology, which won't allow the various services to run as HA.
+
+To achieve the desired topology described in this enhancement,OpenShift cluster
+administrators can only deploy a 4-nodes clusters by first deploying a compact
+cluster (3 schedulable masters) and then adding a host as a day 2 operation.
+Another option is for them to This provides the same results as aimed in this
+enhancement but with a poor and slower user experience.
+
+Today, because of the current limitations in the installer, it takes
+administrators significantly longer to achieve the desired topology, when only
```

Comment:
Significantly longer? Is this longer in terms of the time that an administrator has to be actively doing something? Or is this longer in terms of wall clock time before an administrator has the cluster configured in the way that they want?

Reply:
It becomes 2 step, rather than 1. 
Timewise, it's basically a delta between installing a clusters and adding a worker to it - of course, on BM, it takes longer  - server boot time - 3 reboots if I recall that in case of single installation will happen almost in parallel (1 reboot in parallel to the control plane, and 2 only after the api moves to the control plane to server the ignition.

Reply:
I suppose what I am caught up on is the "significantly" part. Can we remove that color unless we are providing specifics to define what we deem "significant"?

Reply:
> I suppose what I am caught up on is the "significantly" part. Can we remove that color unless we are providing specifics to define what we deem "significant"?

Sure, I will replace it with `it takes administrators at >2h longer` and remove the significantly

Reply:
I don't want to get caught up on this detail, but I am dumbfounded as to why it would take more than 2 hours to add a worker to a cluster. There are a number of ways to accomplish this via one-line change to installer manifests. And even if you wait until the cluster is installed, the change should be small to make and the rollout of the machines should not take too long. Maybe I am missing something in the context of ZTP that adds to the burden, as I am more used to cloud platforms.